python/lsst/ctrl/mpexec/execFixupDataId.py

```diff
@@ -60,52 +60,24 @@ def assoc_fixup():
     dimensions : `str` or sequence [`str`]
         One or more dimension names, quanta execution will be ordered
         according to values of these dimensions.
-    reverse : `bool`, optional
-        If `False` (default) then quanta with higher values of dimensions
-        will be executed after quanta with lower values, otherwise the order
-        is reversed.
     """
 
     def __init__(self, taskLabel: str, dimensions: Union[str, Sequence[str]], reverse: bool = False):
         self.taskLabel = taskLabel
         self.dimensions = dimensions
-        self.reverse = reverse
         if isinstance(self.dimensions, str):
             self.dimensions = (self.dimensions, )
         else:
             self.dimensions = tuple(self.dimensions)
 
-    def _key(self, qdata: QuantumIterData) -> Tuple[Any, ...]:
-        """Produce comparison key for quantum data.
-
-        Parameters
-        ----------
-        qdata : `QuantumIterData`
-
-        Returns
-        -------
-        key : `tuple`
-        """
-        dataId = qdata.quantum.dataId
-        key = tuple(dataId[dim] for dim in self.dimensions)
-        return key
-
-    def fixupQuanta(self, quanta: Iterable[QuantumIterData]) -> Iterable[QuantumIterData]:
-        # Docstring inherited from ExecutionGraphFixup.fixupQuanta
-        quanta = list(quanta)
-        # Index task quanta by the key
-        keyQuanta = {}
-        for qdata in quanta:
-            if qdata.taskDef.label == self.taskLabel:
-                key = self._key(qdata)
-                keyQuanta.setdefault(key, []).append(qdata)
-        if not keyQuanta:
+    def fixupQuanta(self, graph: QuantumGraph) -> QuantumGraph:
+        taskDef = graph.findTaskDefByLabel(self.taskLabel)
+        if taskDef is None:
             raise ValueError(f"Cannot find task with label {self.taskLabel}")
-        # order keys
-        keys = sorted(keyQuanta.keys(), reverse=self.reverse)
-        # for each quanta in a key add dependency to all quanta in a preceding key
-        for prev_key, key in zip(keys, keys[1:]):
-            prev_indices = frozenset(qdata.index for qdata in keyQuanta[prev_key])
-            for qdata in keyQuanta[key]:
-                qdata.dependencies |= prev_indices
-        return quanta
+        quanta = list(graph.quantaForTask(taskDef))
```

Comment:
What order do they come out? Previous version ordered them by dataId, can't we do the same here?

Reply:
Added back in